### ISSUE TITLE
How to increase the accuracy of HRNET model ??

### ISSUE THREAD

**Issue by YuktiADY** [open-mmlab/mmpose#1403]
Hallo,

I have been  training a higher resolution model hrnet_w48_384 X 288 (without using pretrained model) with these specifications below. Addded albumentations as well with Gauss Noise and training on COCO and my custom dataset  for around 50 epochs.

After training the hrnet_w32_384x288 results are AP = 0.7668 , before training ( AP on test dataset) AP = 0.801. AP is not getting  better than 0.801. 

Please find my config below:

optimizer = dict(
type='Adam',
lr=5e-4,
)
optimizer_config = dict(grad_clip=None)
learning policy

lr_config = dict(
policy='poly',
power=0.9,
min_lr=3e-5,
by_epoch=True)
total_epochs =50

Please suggest how can i increase accuracy because i have tried training lower and higher resolution models as well.

**Comment by jin-s13** [comment]
I do not quite understand your question. 
Just to confirm, 
(1) do you train hrnet_w48_384 X 288 or hrnet_w32_384 X 288?
(2) By default, we will train 210 epochs, why do you only train 50 epochs?
(3) How do you train on COCO and the custom dataset? Is your custom dataset also follows the coco human 17-keypoint definition? 
(4) It reads that you did not use pre-trained model, then what do you mean by  `before training ( AP on test dataset) AP = 0.801`? Is it the performance of the pre-trained model?

**Comment by YuktiADY** [comment]
> I do not quite understand your question. Just to confirm, (1) do you train hrnet_w48_384 X 288 or hrnet_w32_384 X 288? (2) By default, we will train 210 epochs, why do you only train 50 epochs? (3) How do you train on COCO and the custom dataset? Is your custom dataset also follows the coco human 17-keypoint definition? (4) It reads that you did not use pre-trained model, then what do you mean by `before training ( AP on test dataset) AP = 0.801`? Is it the performance of the pre-trained model?

1.  I have trained both, lets consider now hrnet_w348 X 288.
2. Yes 210 epochs is maximum, but its not necesaary to train only for 210 epochs , we can start training for less if the learning rate comes to be flat we can stop the training .
3. Intially I was training the model on my custom dataset but the AP and AR starts deteriorating with every epochs. So, i concatenated the COCO and Theodore dataset and trained on that dataset.
 Yes it follows annotations as COCO.
ALso i added ALbumentation with Gauss Noise
4. I meant in order to select the best model to train as we have  Resnet (higher resolution) and HRNET (higher, lower resolution) I first tested all the model on my test dataset before training  and one which gave best AP selected that to finetune and train that particular model.
So, AP before training on Test Dataset = 0.801
     AP after training on COCO + Theodore and validated on Test dataset results in = 0.7668 or 0.77 ,It doesnot crosss 0.77 AP.
      Like after training it should reach above 0.801 or around it atleast.

No ,its not the performance of pretrained model ( in load_from i am not using the pretrained model because when i used it the results were not better so i didnt use it and training from scratch)
 
Alsp please let me know if any part is unclear from my end.   
Please find my config file changes for your reference which i did:

load_from = None
#load_from = 'https://download.openmmlab.com/mmpose/top_down/hrnet/hrnet_w48_coco_384x288-314c8528_20200708.pth' 
resume_from = '/home/yukti/mmpose/theodore_2022-05-24/epoch_50.pth'
#resume_from = None
dist_params = dict(backend='nccl')
workflow = [('train', 1)]
checkpoint_config = dict(interval=5)
evaluation = dict(interval=1, metric='mAP', save_best='AP')

optimizer = dict(
    type='Adam',
    lr=5e-4,
)
optimizer_config = dict(grad_clip=None)
# learning policy
lr_config = dict(
    policy='poly',
    power=0.9, 
    min_lr=3e-5, 
    by_epoch=True)
total_epochs = 60
log_config = dict(
    interval=50,
    hooks=[
        dict(type='TextLoggerHook'),
        dict(type='TensorboardLoggerHook')
    ])

# Added data augmentation 
dict(
        type='Albumentation',
        transforms=[
            dict(
                type='GaussNoise',
                var_limit=(10.0, 50.0)),
        ]),

 val=dict(
        type=dataset_type,
        #ann_file=f'{data_root}/coco_annotations/person_keypoints_valid.json',
        ann_file=f'/mnt/dst_datasets/own_omni_dataset/FES_keypoints/scenario2/person_keypoints_scenario2.json',
        img_prefix=f'/mnt/dst_datasets/own_omni_dataset/FES_keypoints/scenario2/JPEGImages/',
        #img_prefix=f'{data_root}/valid/img_png/',
        data_cfg=data_cfg,
        pipeline=val_pipeline,
        dataset_info={{_base_.dataset_info}}),
    test=dict(
        type=dataset_type,
        #ann_file=f'{data_root}/coco_annotations/person_keypoints_valid.json',
        ann_file=f'/mnt/dst_datasets/own_omni_dataset/FES_keypoints/scenario2/person_keypoints_scenario2.json',
        #img_prefix=f'{data_root}/valid/img_png/',
        img_prefix=f'/mnt/dst_datasets/own_omni_dataset/FES_keypoints/scenario2/JPEGImages/',
        data_cfg=data_cfg,
        pipeline=test_pipeline,
        dataset_info={{_base_.dataset_info}}),

**Comment by YuktiADY** [comment]
Awaiting for your response

**Comment by jin-s13** [comment]
1. Are you using the Theodore dataset? Do you mean this synthetic one? https://arxiv.org/pdf/2011.05719v1.pdf
I can not find the official website of this dataset. Are you sure how the keypoints are defined? Since they are from different domains, the annotations of COCO and Theodore may be different in some way (variance or something).
In this case, I will suggest using a model with 34 output channels, where the first 17 channels for training coco dataset, and the remaining for Theodore dataset. 
Step1: train the model with the combination of COCO+Theodore, 
Step2: finetune on Theodore only. 
In general, this will produce better results. 
2. `we can start training for less if the learning rate comes to be flat we can stop the training .`
We should monitor the accuracy instead of learning rate, aren't we?
3. Since your test set does not have gaussian noises, I think ALbumentation with Gauss Noise is not necessary. 
4. You may also try Pretraining + lower learning rate
